battles/views.py

```diff
@@ -81,3 +92,7 @@ def get_context_data(self, **kwargs):
 
     def get_success_url(self):
         return reverse('battles:details', kwargs={'pk': self.kwargs['pk']})
+
+    def form_valid(self, form):
+        form.save()
+        return super().form_valid(form)
```

Comment:
Did you considered inherit from a `CreateView`? Cause you wouldn't need to override the `form_valid`